### ISSUE TITLE
Refactor the header files in wdl/include to accommodate the wdl_common.h file

### ISSUE THREAD

**Issue by mewthew** [uchicago-cs/chiventure#334]
We have added a wdl_common.h file to help simplify dependencies among the header files. Some of these header files have not been changed to coincide with the wdl_common.h file. This issue aims to organize the #include statements in the header files with the addition of the wdl_common.h file. These changes will be done in the wdl/load_wdl branch.

**Comment by mewthew** [comment]
Update:
- The header files that need to be updated are as follows: load_game, load_room, load_item, validate, and parse.h
- I will also be updating the corresponding c files to the header files such that they only include the corresponding h file

**Comment by mewthew** [comment]
Update:
- The h files and c files have been updated to comply with the wdl_common file

Next Steps:
- I've realized that the test_validation file is not passing because we have not defined a file path to a YAML file, which I will fix by defining a file path within the test_validation file

**Comment by mewthew** [comment]
Update:
- The test_validation error has been fixed via commit: cac6326

**Comment by mewthew** [comment]
I have refactored the header files (load_wdl, load_item, load_room, load_game, validate, and parse) to accommodate the wdl_common.h file and also updated their corresponding c files. I found a small bug causing the test_validation tests to fail due to an undefined file path to a yaml file, which I also have addressed within this issue. Now, each module has been updated in a more organized manner through the wdl_common.h file.
